src/components/input/select/index.tsx

```diff
@@ -179,12 +179,8 @@ export class SmoothlyInputSelect implements Input, Editable, Clearable, Componen
 		event && event.stopPropagation()
 		const wasButtonClicked =
 			event &&
-			event
-				.composedPath()
-				.some(
-					element =>
-						"tagName" in element && typeof element.tagName === "string" && this.buttonTagNames.includes(element.tagName)
-				)
+			event.composedPath().includes(this.iconsDiv as EventTarget) &&
```

Comment:
change the first composedPath call to have a question mark `event.composedPath()` -> `event?.composedPath()` That was we can remove the `event &&`

Reply:
and maybe we combine best of both worlds. lets use your some call so we can avoid the casts
```ts
event.composedPath().some(e => e == this.iconsDiv) && !event.composedPath().some(e => e == this.toggle)
```